packages/playground/src/ui.ts

```diff
@@ -109,22 +141,11 @@ export async function createUI(host: BrowserHost) {
       swaggerOutputFile: "cadl-output/openapi.json",
       emitters: ["@cadl-lang/openapi3"],
     });
-    const markers: monaco.editor.IMarkerData[] = [];
-    for (const diag of program.diagnostics) {
-      const loc = getSourceLocation(diag.target);
-      if (!loc) continue;
-      const start = loc.file.getLineAndCharacterOfPosition(loc.pos);
-      const end = loc.file.getLineAndCharacterOfPosition(loc.end);
-
-      markers.push({
-        startLineNumber: start.line + 1,
-        startColumn: start.character + 1,
-        endLineNumber: end.line + 1,
-        endColumn: end.character + 1,
-        message: diag.message,
-        severity: monaco.MarkerSeverity.Error,
-      });
-    }
+    const markers: monaco.editor.IMarkerData[] = program.diagnostics.map((diag) => ({
+      ...getMarkerLocation(diag.target),
+      message: diag.message,
+      severity: monaco.MarkerSeverity.Error,
+    }));
```

Comment:
I'm missing (before and after) which file is impacted. What happens if the error is in an import, does it squiggle in the open document at a seemingly random location? Or am I just missing where this is handled elsewhere?

Reply:
hhm, I guess the only cases were this would happen is if we have an error that shows up in the rest library or another library as it doesn't have multi file support. In that case I would suspect it would just show in a bad location in this file looking at it. I can double check

Reply:
Probably worth being defensive about that since we still have issues sometimes where we report consumption errors against the library declaration.

Reply:
No need to tie it to this PR, though.

Reply:
Yeah shows up in the the current file at the position where it is in the other file 
![image](https://user-images.githubusercontent.com/1031227/161338127-c3ddc0ba-7552-434e-ad68-dfdbf4c109b5.png)

Could potentially just reset to 1,1 if it is not in the main file as an easy fix